common/params.go

```diff
@@ -96,16 +97,16 @@ type metadataValidator interface {
 }
 
 type RequireMetadata struct {
-	Expected []string
+	ExpectedMetadataKeys []string
 }
 
-func (RequireMetadata) validateMetadata(parameters Parameters) error {
+func (m RequireMetadata) validateMetadata(parameters Parameters) error {
 	if parameters.Metadata == nil {
 		return ErrMissingMetadata
 	}
 
-	for _, key := range parameters.Metadata {
-		if key == "" {
+	for _, key := range m.ExpectedMetadataKeys {
```

Comment:
Thanks for catching this! Might be a good idea to make this case insensitive and add an comment saying that in the struct